modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/APMMetric.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.apm.internal.metrics;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.telemetry.metric.DoubleCounter;
+import org.elasticsearch.telemetry.metric.DoubleGauge;
+import org.elasticsearch.telemetry.metric.DoubleHistogram;
+import org.elasticsearch.telemetry.metric.DoubleUpDownCounter;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongGauge;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.LongUpDownCounter;
+import org.elasticsearch.telemetry.metric.MetricName;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.elasticsearch.telemetry.apm.internal.APMAgentSettings.APM_ENABLED_SETTING;
+
+public class APMMetric extends AbstractLifecycleComponent implements org.elasticsearch.telemetry.metric.Metric {
+    private static final Meter NOOP = OpenTelemetry.noop().getMeter("noop");
+    private final Instruments instruments = new Instruments(NOOP);
+    private volatile boolean enabled;
+    private final AtomicReference<APMServices> services = new AtomicReference<>();
+
+    record APMServices(Meter meter, OpenTelemetry openTelemetry) {}
+
+    public APMMetric(Settings settings) {
+        this.enabled = APM_ENABLED_SETTING.get(settings);
```

Comment:
We should call setEnabled here. currently a default value for `tracing.apm.enabled` is false, so there will always be a cluster state update event that will update the value to true.
However, if we at some point change the default to true, there will be no setting value change, so `setEnabled` will never be called from cluster settings updater https://github.com/elastic/elasticsearch/blob/main/modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/APMAgentSettings.java#L45

I think we can mark the class final, so that we can safely call setEnabled (it is a public method, so there is a risk of `this escape` problem if someone extend this class).